src/meshlink.c

```diff
@@ -713,173 +547,91 @@ static bool finalize_join(meshlink_handle_t *mesh) {
 		return false;
 	}
 
-	char filename[PATH_MAX];
-	snprintf(filename, sizeof(filename), "%s" SLASH "meshlink.conf", mesh->confbase);
-
-	FILE *f = fopen(filename, "w");
-
-	if(!f) {
-		logger(mesh, MESHLINK_DEBUG, "Could not create file %s: %s\n", filename, strerror(errno));
+	if(!count) {
+		logger(mesh, MESHLINK_ERROR, "Incomplete invitation file!\n");
 		return false;
 	}
 
-	fprintf(f, "Name = %s\n", name);
-
-	// Wipe all old host config files and invitations
-	snprintf(filename, sizeof(filename), "%s" SLASH "hosts", mesh->confbase);
-	deltree(filename);
-
-	if(mkdir(filename, 0777) && errno != EEXIST) {
-		logger(mesh, MESHLINK_DEBUG, "Could not create directory %s: %s\n", filename, strerror(errno));
+	// Initialize configuration directory
+	if(!config_init(mesh)) {
 		return false;
 	}
 
-	snprintf(filename, sizeof(filename), "%s" SLASH "invitations", mesh->confbase);
-	deltree(filename);
+	// Write main config file
+	uint8_t outbuf[4096];
+	packmsg_output_t out = {outbuf, sizeof(outbuf)};
+	packmsg_add_uint32(&out, MESHLINK_CONFIG_VERSION);
+	packmsg_add_str(&out, name);
+	packmsg_add_bin(&out, ecdsa_get_private_key(mesh->private_key), 96);
+	packmsg_add_uint16(&out, atoi(mesh->myport));
 
```

Comment:
I think writing main config file, writing host config file etc., type of code should be modular. As the loading and unloading of the package message following a format I think we should write it in a function and can reuse in the code in both `meshlink_setup()` and `finalize_join()` which uses the same code.

Reply:
Good point, I'll add such a function.